craft_parts/packages/snaps.py

```diff
@@ -134,18 +134,22 @@ def get_store_snap_info(self) -> Optional[Dict[str, Any]]:
                 try:
                     self._store_snap_info = _get_store_snap_info(self.name)
                     break
-                except exceptions.HTTPError as http_error:
-                    logger.debug(
-                        "The http error when checking the store for %s is %d "
-                        "(retries left %d)",
-                        self.name,
-                        http_error.response.status_code,
-                        retry_count,
-                    )
-                    if http_error.response.status_code == 404:  # noqa: PLR2004
-                        raise errors.SnapUnavailable(
-                            snap_name=self.name, snap_channel=self.channel
-                        ) from http_error
+                except exceptions.HTTPError as http_err:
+                    if http_err.response:
+                        logger.debug(
+                            "The http error when checking the store for %s is %d "
+                            "(retries left %d)",
+                            self.name,
+                            http_err.response.status_code,
+                            retry_count,
+                        )
+                        if http_err.response.status_code == 404:  # noqa: PLR2004
+                            raise errors.SnapUnavailable(
+                                snap_name=self.name, snap_channel=self.channel
+                            ) from http_err
+                    else:
+                        logger.debug("No response information in http error.")
```

Comment:
I think we should add the rest of the original debug message here (like self.name and retry_count)

Reply:
why was this changed? seems unrelated to pydantic

Reply:
I carried over from Claudio's experimental branch for pydantic2. I suppose he was getting some http_errors without responses. I didn't test if that's still needed but an extra check that causes no harm.